pkg/gardenlet/operation/botanist/clusterautoscaler.go

```diff
@@ -97,13 +97,17 @@ func (b *Botanist) calculateMaxNodesForPodsNetwork(shoot *gardencorev1beta1.Shoo
 		// Calculate how many subnets with nodeCIDRMaskSize can be allocated out of the pod network (with podCIDRMaskSize).
 		// This indicates how many Nodes we can host at max from a networking perspective.
 		var maxNodeCount = &big.Int{}
-		exp := int64(*shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize) - int64(podCIDRMaskSize)
+		exp := 64 - int64(podCIDRMaskSize)
```

Comment:
Where is the 64 coming from?

Reply:
It is the default `NodeCIDRMaskSize`. This is not good. We can't even configure it in a dual-stack shoot.
Actually, we should get `NodeCIDRMaskSize` from the infrastructure. Something we need to change.

Reply:
Offline discussion lead to a more explicit comment explaining the situation.